core/src/main/resources/reference.conf

```diff
@@ -44,6 +44,12 @@ akka.kafka.consumer {
   # messages already in the stream (required for successful committing).
   # Prefer use of `DrainingControl` over a large stop-timeout.
   stop-timeout = 30s
+
+  # The timeout waiting for the transactional partitioned streams that close
+  # on every rebalance when revoking the partitions.
+  # This timeout is used in a blocking operation and will suspend consumption
+  # until either it times out, or the partitioned streams complete.
+  eos-stream-stop-timeout = 40s
```

Comment:
How does this timeout relate to eg. `session.timeout.ms` and `heartbeat.interval.ms`? 
If the consumer is blocked for the whole timeout, at what values would the broker consider it dead?

Reply:
I don't see anything obvious that aborts the call to the revoke handler.  It seems to be called synchronously as part of the poll, which makes sense since the `ConsumerRebalanceListener` javadocs says these callbacks are made as part of the user poll thread.  There's a timer with an infinite (`Long.MAX`) duration that is started before it's invoked.

Reply:
It looks like the Consumer Group Coordinator process on the broker determines the timeout.  `group.max.session.timeout.ms` has a default of 300 seconds.

Reply:
Config was removed.